website/docs/r/device_security_group.html.markdown

```diff
@@ -0,0 +1,112 @@
+---
+subcategory: "Security Center"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_device_security_group"
+description: |-
+  Manages a Device Security Group.
+---
+
+# azurerm_device_security_group
+
+Manages a Device Security Group.
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  name     = "example-resources"
+  location = "West Europe"
+}
+
+resource "azurerm_iothub" "example" {
+  name                = "example-IoTHub"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+  sku {
+    name     = "S1"
+    capacity = "1"
+  }
+}
+
+resource "azurerm_iot_security_solution" "example" {
+  name                = "example-Iot-Security-Solution"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+  display_name        = "Iot Security Solution"
+  iothub_ids          = [azurerm_iothub.example.id]
+}
+
+resource "azurerm_device_security_group" "example" {
+  name               = "example-device-security-group"
+  target_resource_id = azurerm_iothub.example.id
+
+  allow_list_rule {
+    type   = "LocalUserNotAllowed"
+    values = ["user0"]
+  }
+
+  time_window_rule {
+    type             = "ActiveConnectionsNotInAllowedRange"
+    min_threshold    = 0
+    max_threshold    = 30
+    time_window_size = "PT5M"
+  }
+
+  depends_on = [azurerm_iot_security_solution.example]
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) Specifies the name of the Device Security Group. Changing this forces a new resource to be created.
+
+* `target_resource_id` - (Required) The ID of the Azure Resource which to create the Device Security Group. Changing this forces a new resource to be created.
```

Comment:
what sort of target resources are valid here? only iot hubs? in which case
```suggestion
* `target_iot_hub_id` - (Required) The ID of the IoT Hub which to link the Device Security Group to. Changing this forces a new resource to be created.
```
 (also grammer)